app/workers/education_form/create_daily_fiscal_year_to_date_report.rb

```diff
@@ -234,7 +234,7 @@ def generate_csv
     def perform
       Sentry::TagRainbows.tag
       filename = generate_csv
-      return unless FeatureFlipper.send_email? && !Settings.hostname.include?('dev-api')
+      return unless FeatureFlipper.send_email? && !Settings.hostname.include?('vetsgov-internal')
```

Comment:
This check is temporary I'm guessing because of FeatureFlipper, but I strongly advise against gating in anyway off of hostname. The handful of places that we do this are related only to UrlService and this could in theory change in the future. 

Instead I advise creating your own whitelist of available hosts and checking against request.host where possible, but even then we generally prefer to use custom Settings.yml for each service to say if it is enabled or not.

Reply:
Gating on hostname makes things a little bit more fragile than we'd like, and it also runs the potential of introducing code that looks safe, but isn't always the case if its not handled correctly. One could inadvertently introduce logic that appears safe but isn't mindful of how easy it is to spoof a hostname.

Reply:
@saneshark : The only thing I am trying to do here is avoid bunch and bunch of emails being sent from review instances for one of the sidekiq job. I couldn't find any other way from the rails.env or profile to identify if the system is running on review vs dev vs staging vs prod environment. There is no point introducing a new settings for a need to send or not send email for these education form reports, hence this change. open to any suggestion

Reply:
The review instances have their own settings.yml file I believe. Dev/Staging/Review/Prod all have unique settings.yml overrides. You should just be able to set the feature flag to false there.

https://github.com/department-of-veterans-affairs/devops/tree/master/ansible/deployment/config/vets-api

Let me see if I can find the exact file you need to change.

Reply:
@lihanli you probably have a bit more context around this. I totally understand what and why @abhijitvk is doing it this way, my only concern is that we might be introducing a new paradigm of how certain features and/or services might be gated outside the normal way of doing it. Using Setting.hostname is convenient, and I respect that, and I don't want to slow down development. My only concern is that we might introduce a new precedent that others will copy/paste from that is outside of our normal use-case. Is it possible to instead maybe disable the cron jobs for the sidekiq workers in the review instances and achieve the same result?

Reply:
hostname does seem fragile, it's possible in the future a non-review instance server will also have a `vetsgov-internal` hostname

we've traditionally used feature flags to control these things but @wyattwalter mentioned he isn't opposed to adding something to settings that would specify the environment (dev, staging, review, etc), maybe we could try that?

Reply:
Alright. I made updates based on the feedback. Added a new feature flag for reports namely `reports.env` . Please review and confirm if this is acceptable.

Reply:
I have also created a devops issue requesting changes for the settings yml files. https://github.com/department-of-veterans-affairs/devops/issues/4133

Reply:
I don't mind having an overall variable to tell us what environment the app is running in, but putting an `env` variable inside the `reports` namespace seems a little odd. I would argue for  doing one or the other: a top level one for knowing the environment, or be direct about it in the `reports` namespace, as in `Settings.reports.send_email` as a boolean.

Reply:
Thanks @wyattwalter. Just to avoid going back and forth on this, please suggest specific property that I can add to tell what environment app is running. 
`vets-api.env` or `env`. 

let me know and I can make final changes accordingly.

Reply:
added overall variable as `api.env` to show what environment app is running in